src/datepicker/timepicker.js

```diff
@@ -54,15 +52,73 @@ function secondsToLabel(seconds, format = '12') {
   return `${zeroPrefix(hours)}:${zeroPrefix(minutes)}`;
 }
 
+/**
+ * Converts a time string, e.g. 10:00, to one or more possible TimePicker
+ * options representing that time.
+ */
+function stringToOptions(str, format): Array<OptionT> {
+  // leading zero is optional, AM/PM is optional
+  const twelveHourRegex = /^(1[0-2]|0?[1-9]):([0-5][0-9]) ?([AaPp][Mm]?)?$/;
```

Comment:
I'm not quite sure what's going on in the examples below, but it would be great if we can refine this regex to handle some of the strange cases below. It's not that bad, and would be happy to merge as-is if we can't figure it out.

<img width="327" alt="Screen Shot 2019-05-29 at 9 17 42 AM" src="https://user-images.githubusercontent.com/5317799/58573613-fb517880-81f2-11e9-804d-b677380b567b.png">
<img width="327" alt="Screen Shot 2019-05-29 at 9 17 31 AM" src="https://user-images.githubusercontent.com/5317799/58573614-fb517880-81f2-11e9-9b02-39e05c14fbd0.png">
<img width="327" alt="Screen Shot 2019-05-29 at 9 17 10 AM" src="https://user-images.githubusercontent.com/5317799/58573615-fbea0f00-81f2-11e9-816a-1c5c9b3ef559.png">

Reply:
Hmm, I noticed this as well... These search behaviors seem to replicate on the existing timepicker as well, I just tested it on the doc site. Definitely something I'd like to figure out how to fix, but I don't think it relates to changes in this PR.

Reply:
sounds good to me. we can look into this later